src/Symfony/Component/HttpClient/Response/HttplugPromise.php

```diff
@@ -54,6 +54,12 @@ public function getState(): string
      */
     public function wait($unwrap = true)
     {
-        return $this->promise->wait($unwrap);
+        $result = $this->promise->wait($unwrap);
+
+        if ($result instanceof HttplugPromiseInterface) {
```

Comment:
Should this be a while loop?

Reply:
while $result is HttplugPromiseInterface? emm... maybe yes!

Reply:
Fixed and added test to ensure arbitrary levels of promises work.